packages/js/src/dashboard/widgets/site-kit-setup-widget.js

```diff
@@ -68,17 +70,26 @@ const useSiteKitConfiguration = ( dataProvider, remoteDataProvider ) => {
  *
  * @param {DataProvider} dataProvider The data provider.
  * @param {RemoteDataProvider} remoteDataProvider The remote data provider.
- * @param {function} onRemove The function to call when the widget is removed.
+ * @param {function} removeWidget The function to remove a widget.
+ * @param {function} addWidget The function to add a widget.
+ * @param {WidgetType[]} siteKitWidgets The site kit widgets to add once connected.
  *
  * @returns {JSX.Element} The widget.
  */
-export const SiteKitSetupWidget = ( { dataProvider, remoteDataProvider, onRemove } ) => {
-	const { config, grantConsent, dismissPermanently } = useSiteKitConfiguration( dataProvider, remoteDataProvider );
+export const SiteKitSetupWidget = ( { dataProvider, remoteDataProvider, removeWidget, addWidget, siteKitWidgets } ) => {
+	const currentWidgetType = siteKitWidgets[ 0 ];
```

Comment:
Honestly, I'm not sure its own ID should live here. I get you did not want to make this widget aware of the WidgetFactory.
But this feels like maybe the WidgetFactory should already have bound the removeWidget to this ID. But that would require state management in the WidgetFactory, which is also meh.

Maybe just separate these own and to add already in the factory?